cmake/Modules/merge_static_libs.cmake

```diff
@@ -1,33 +1,62 @@
-function(merge_static_libs target)
+function(merge_static_libs target unmerged_libs)
   set(args ${ARGN})
 
+  set(dependencies)
+
   foreach(lib ${args})
-    if("${lib}" MATCHES "(\\${CMAKE_STATIC_LIBRARY_SUFFIX}|dav1d\.a)$")
-      list(APPEND libs ${lib})
+    if(TARGET "${lib}")
+      get_target_property(target_type ${lib} TYPE)
+
+      if(${target_type} STREQUAL "STATIC_LIBRARY")
+        list(APPEND libs $<TARGET_FILE:${lib}>)
+        list(APPEND dependencies "${lib}")
+      else()
+        list(APPEND unmerged_libs "${lib}")
+      endif()
+    elseif("${lib}" MATCHES "(\\${CMAKE_STATIC_LIBRARY_SUFFIX}|dav1d\.a)$")
+      list(APPEND libs "${lib}")
+
+      if(EXISTS "${lib}")
+        list(APPEND dependencies "${lib}")
+      endif()
+    else()
+      list(APPEND unmerged_libs "${lib}")
     endif()
   endforeach()
 
+  set(source_file ${CMAKE_CURRENT_BINARY_DIR}/${target}_depends.c)
+  add_library(${target} STATIC ${source_file})
+
+  add_custom_command(
+    OUTPUT ${source_file}
+    COMMAND ${CMAKE_COMMAND} -E echo "const int dummy = 0;" > ${source_file}
+    DEPENDS ${dependencies}
+  )
+
+  add_custom_command(
+    TARGET ${target}
+    POST_BUILD
+    COMMENT "Merge static libraries"
+    COMMAND ${CMAKE_COMMAND} -E remove $<TARGET_FILE:${target}>
+  )
+
   if(APPLE)
     add_custom_command(
       TARGET ${target}
       POST_BUILD
-      COMMENT "Merge static libraries with libtool"
-      COMMAND ${CMAKE_COMMAND} -E rename $<TARGET_FILE:${target}> $<TARGET_FILE:${target}>.tmp
-      COMMAND xcrun libtool -static -o $<TARGET_FILE:${target}> $<TARGET_FILE:${target}>.tmp ${libs}
-      COMMAND ${CMAKE_COMMAND} -E remove $<TARGET_FILE:${target}>.tmp
+      COMMAND xcrun libtool -static -o $<TARGET_FILE:${target}> ${libs}
     )
   elseif(CMAKE_C_COMPILER_ID MATCHES "^(Clang|GNU|Intel|IntelLLVM)$")
     add_custom_command(
       TARGET ${target}
       POST_BUILD
-      COMMENT "Merge static libraries with ar"
-      COMMAND ${CMAKE_COMMAND} -E rename $<TARGET_FILE:${target}> $<TARGET_FILE:${target}>.tmp
       COMMAND ${CMAKE_COMMAND} -E echo CREATE $<TARGET_FILE:${target}> >script.ar
-      COMMAND ${CMAKE_COMMAND} -E echo ADDLIB $<TARGET_FILE:${target}>.tmp >>script.ar
     )
 
     foreach(lib ${libs})
-      add_custom_command(TARGET ${target} POST_BUILD
+      add_custom_command(
+        TARGET ${target}
+        POST_BUILD
```

Comment:
Did you run cmake-format on that file, as indicated in thee README? Thx

Reply:
Oh, I overlooked that. I made one commit that incorporated your feedback, and then a second commit where a ran cmake-format on CMakeLists.txt and merge_static_libs.cmake

Reply:
Thx! Can you please rebase? I just ran it with https://github.com/AOMediaCodec/libavif/pull/1692. This way, it will remove some noise. Otherwise, LGTM

Reply:
Done.